src/orgasm_control.c

```diff
@@ -343,10 +346,17 @@ static void orgasm_control_updateEdgingTime() { // Edging+Orgasm timer
         // Detect if within post orgasm session
         if ((esp_timer_get_time() / 1000UL) < (post_orgasm_state.post_orgasm_start_millis +
                                                post_orgasm_state.post_orgasm_duration_millis)) {
-            output_state.motor_speed = Config.motor_max_speed;
+            // continue to raise motor to max speed
+            if (output_state.motor_speed <= (Config.motor_max_speed - output_state.motor_increment)
+                ) {
+                update_check(output_state.motor_speed, output_state.motor_speed + output_state.motor_increment);
+            } else {
+                update_check(output_state.motor_speed, Config.motor_max_speed );
+            }
```

Comment:
All of these code 

plus this code
 `output_state.motor_increment = calculate_increment(Config.motor_start_speed, Config.motor_max_speed, 
Config.motor_ramp_time_s);`

plus this codes
`    // Control output while motor control is paused
    if (output_state.control_motor == OC_MANUAL_CONTROL) {
        uint8_t speed = orgasm_control_get_motor_speed();
        _set_speed(speed);
    }`

could be reduced in one function and one additional structure. For example 
`
static struct {
    float motor_start_speed;
    float motor_max_speed;
    float motor_ramp_time_s;
} config_state;

/**
 * @brief Simplified method to increment motor speed up to a config max
 * @The increment is calculated at first execution and recalculated if config changes
 * @param void
 */
void increment_motor()
if ( config_state.motor_max_speed != Config.motor_max_speed ||
        config_state.motor_start_speed != Config.motor_start_speed ||
        config_state.motor_ramp_time_s != Config.motor_ramp_time_s) {
   config_state.motor_max_speed != Config.motor_max_speed;
   config_state.motor_start_speed!= Config.motor_start_speed;
   config_state.motor_ramp_time_s!= Config.motor_ramp_time_s;   
   output_state.motor_increment = calculate_increment(Config.motor_start_speed, Config.motor_max_speed, Config.motor_ramp_time_s);   
}
if (output_state.motor_speed <= (Config.motor_max_speed - output_state.motor_increment)) {
     update_check(output_state.motor_speed, output_state.motor_speed + output_state.motor_increment);
} else {
      update_check(output_state.motor_speed, Config.motor_max_speed );
}
_set_speed(output_state.motor_speed);`


I wound call increment_motor() in the Pre-orgasm part and the Post-orgasm part and delete all the listed code